include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -226,7 +234,7 @@ def BufferOp : Handshake_Op<"buffer", [NoSideEffect, HasClock,
     TypeAttr:$dataType,
     Confined<I32Attr, [IntMinValue<1>]>:$size,
     AnyType:$operand,
-    BoolAttr:$sequential,
+    SequentialStrEnum:$sequential,
```

Comment:
To keep things general, it might be better to rename things to `BufferTypeEnum` and `$bufferType`. I think this will make reading the code that references this attribute a bit easier (i.e. `bufferOp->bufferType() == "fifo"` makes more sense than `bufferOp->sequential() == "fifo"`).